Change bundle ID for WebKit extensions

pvollan opened the pull request:
#### 96bf3e7cda56239b63932e75080cda86c08c03d7
<pre>
Change bundle ID for WebKit extensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=267092">https://bugs.webkit.org/show_bug.cgi?id=267092</a>
<a href="https://rdar.apple.com/120485965">rdar://120485965</a>

Reviewed by Elliott Williams.

Change bundle ID for WebKit extensions to match the XPC service variants.
This will make sure existing WebKit bundle ID checks still succeed.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/GPUExtension.xcconfig:
* Source/WebKit/Configurations/NetworkingExtension.xcconfig:
* Source/WebKit/Configurations/WebContentCaptivePortalExtension.xcconfig:
* Source/WebKit/Configurations/WebContentExtension.xcconfig:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
(WebKit::serviceNameAndIdentifier):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/272766@main">https://commits.webkit.org/272766@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e81bc59690b7839cc7acee91600d62f4dea537d7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33011 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11784 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34914 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/35653 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/29830 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/14123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8953 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/35653 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33485 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/14123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/29494 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/35653 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/14123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/36987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/14123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/29861 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/36987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/8949 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/6887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/36987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/10640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4243 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/9580 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Configurations/GPUExtension.xcconfig:
```diff
@@ -25,6 +25,7 @@
 INFOPLIST_KEY_CFBundleDisplayName = GPUExtension;
-PRODUCT_BUNDLE_IDENTIFIER = com.apple.WebKit.GPUExtension;
-PRODUCT_BUNDLE_EXECUTABLE = GPUExtension;
+PRODUCT_NAME = com.apple.WebKit.GPU;
```
Does this actually work? The binary will still end in `.appex`, so it's a different name than the XPC services.

Would it instead be sufficient to change the process name at runtime to match the status quo names, similar to what we do in `WebProcess::updateProcessName`?

pvollan replied:
Yes, I verified that this works. However, I have reduced the patch to just change the bundle ID, since that is the main objective of the patch.

Thanks for reviewing!

emw-apple commented on Source/WebKit/Scripts/process-entitlements.sh:
```diff
@@ -361,7 +361,7 @@ function ios_family_process_webcontent_shared_entitlements()
     plistbuddy Add :com.apple.private.webkit.use-xpc-endpoint bool YES
     plistbuddy Add :com.apple.runningboard.assertions.webkit bool YES
-if [[ "${PRODUCT_NAME}" != WebContentExtension ]]; then
+if [[ "${PRODUCT_BUNDLE_NAME}" != WebContentExtension ]]; then
```
Why did we define PRODUCT_BUNDLE_NAME? AFAICT it's not a standard Xcode build setting.

I see that it's being used to set `CFBundleName` in the extension `Info.plist`s, but in the XPC service targets, the bundle name is the fully-qualified bundle identifier.

pvollan replied:
You are right, I think we can simplify this to match the XPC service targets. There is no need to explicitly define PRODUCT_BUNDLE_NAME. Are you ok with doing this in a follow-up, since it will not actually change the CFBundleName?

emw-apple replied:
That's fine with me!

pvollan replied:
Thanks for approving!

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18019,14 +18019,14 @@
 			inputPaths = (
 			);
-			name = "Copy extension (temporary workaround)";
+			name = "Update extensions";
```
Could you find a more descriptive name? And is this no longer a temporary workaround?

From what I can tell, the script replaces `EXExtensionPointIdentifier` strings from the default values provided in Info.plist, but only during production builds (when DSTROOT exists and extension have been installed to it). I'm a little curious why these can't be hard-coded in Info.plist. Do they need to be different between debug and production builds?

pvollan replied:
That's a good point. I have updated the description to be more informative. This is still a temporary workaround, which I aim to remove soon. The patch changed the install location, so the copy step of the script is no longer needed.

Thanks for reviewing!

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18909,7 +18909,9 @@
-			platformFilter = ios;
+			platformFilters = (
+				ios,
+			);
```
Seems like noise, maybe on account of your Xcode version being newer. Could you delete?

pvollan replied:
Done!

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 55;
```
Ditto.

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18925,19 +18927,25 @@
-			platformFilter = ios;
+			platformFilters = (
+				ios,
+			);
```
Ditto.

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18925,19 +18927,25 @@
-			platformFilter = ios;
+			platformFilters = (
+				ios,
+			);
```
Ditto.

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18925,19 +18927,25 @@
-			platformFilter = ios;
+			platformFilters = (
+				ios,
+			);
```
Ditto.

pvollan committed:
Change bundle ID for WebKit extensions
https://bugs.webkit.org/show_bug.cgi?id=267092
rdar://120485965

Reviewed by Elliott Williams.

Change bundle ID for WebKit extensions to match the XPC service variants.
This will make sure existing WebKit bundle ID checks still succeed.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/GPUExtension.xcconfig:
* Source/WebKit/Configurations/NetworkingExtension.xcconfig:
* Source/WebKit/Configurations/WebContentCaptivePortalExtension.xcconfig:
* Source/WebKit/Configurations/WebContentExtension.xcconfig:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
(WebKit::serviceNameAndIdentifier):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/272766@main
